pkg/apis/core/validation/validation.go

```diff
@@ -1824,6 +1827,19 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 	} else if spec.VolumeMode != nil && !supportedVolumeModes.Has(string(*spec.VolumeMode)) {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("volumeMode"), *spec.VolumeMode, supportedVolumeModes.List()))
 	}
+
+	if spec.DataSource != nil && !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSnapshotDataSource) {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("dataSource"), "VolumeSnapshotDataSource is disabled by feature-gate"))
+	} else if spec.DataSource != nil {
+		if len(spec.DataSource.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("dataSource", "name"), ""))
+		} else if !supportedDataSourceKinds.Has(string(spec.DataSource.Kind)) {
```

Comment:
When you add a second kind or group, you probably want to test them as tuples :)

Reply:
Sure.  supportedDataSourceKinds is defined as sets.NewString(string("VolumeSnapshot”)).  If we added support for a new kind, we can add it to the sets.